.github/scripts/generate_ci_workflows.py

```diff
@@ -155,12 +162,20 @@ def assert_valid(self) -> None:
         if self.arch == 'windows':
             assert self.test_runner_type in WINDOWS_RUNNERS, err_message
 
+    def generate_workflow_name(self) -> str:
+        workflow_name = self.build_environment
+        if not self.on_pull_request:
+            workflow_name = "master-only-" + workflow_name
```

Comment:
cc @walterddr @janeyx99 

self.on_pull_request might be tricky to rely on, previously the context of running `on_pull_request` is easy to understand, however, with the introduction of ciflow, the `on_pull_request` is also set to true for those workflows that can be opt-in at PR time (with the control of labels+probot, but off by default).

Reply:
A side note is that do we really need the workflow name to indicate if it's `master-only`?

Reply:
does it mean "some CI jobs that suppose to be run on master only can be run on PR"?

Reply:
one of the goal of this PR is to display the master-only prefix --> it is hard to distinguish just looking at HUD.. now with this prefix treehugger oncall can immediately clear whether a PR is failing a master-only job thus it is not checked (by default) on PR

Reply:
> does it mean "some CI jobs that suppose to be run on master only can be run on PR"?

Yes, for example, some workflows that are conceptually master-only, they also have `on_pull_request:true`, because we can run them at PR time with `ciflow/slow` or `ciflow/scheduled` labels.

This is done by having them only listening on the events like:

```
on:
  pull_request:
    types: [unassigned]
```

Reply:
oh I think this is fine because --> 
1. generate_ci_workflow.py is used to generate the templates and not dynamically run during PR --> thus the job file name / job name will still have "master-only" prefix.
2. this is just my opinion, but we should still show the name "master-only" for those jobs that were triggered on CI with special labels, because that's the feedback we want to give the developer "because you add a label, we are now running jobs that usually only run on master only".

Reply:
`self.on_pull_request` will basically become `true` for all workflows when ciflow is rolling out to more workflows. I think if that's the case, we need a new field to indicate the difference to name `master-only` instead of just using `self.on_pull_request`

1. A job runs on `master-only` and not on pull_request
2. A job runs on `master-only` by default, but can be enabled on pull_request

Reply:
that sounds like a great follow up item, should I include in this PR or create a follow up later?

Reply:
include in this PR will be great. I also think we can always keep it simple and let each `CIWorkflow` have its own `workflow_name` without computed logic (sometimes duplication is ok if it's designed for readability), it really helps in the future if we are going to just change one workflow_name without worrying about the `__post_init__` logic.